components/director/internal/destinationfetchersvc/syncjob.go

```diff
@@ -21,7 +22,39 @@ type DestinationSyncer interface {
 type SyncJobConfig struct {
 	ElectionCfg       cronjob.ElectionConfig
 	JobSchedulePeriod time.Duration
-	ParallelTenants   int64
+	TenantSyncTimeout time.Duration
+	ParallelTenants   int
+}
+
+func syncSubscribedTenantsDestinations(ctx context.Context, subscribedTenants []string, cfg SyncJobConfig,
+	destinationSyncer DestinationSyncer) int {
+	parallelTenantsSemaphore := semaphore.NewWeighted(int64(cfg.ParallelTenants))
+	wg := sync.WaitGroup{}
+	var syncedTenants int32
```

Comment:
you may use uint32 instead, as there is no way to have negative number of tenants.

Reply:
[`NewWeighted`](https://pkg.go.dev/golang.org/x/sync/semaphore#NewWeighted) requires int64